### PR TITLE
Improve Network Model

### PR DESCRIPTION
**Pull request by mysCod3r**
# Improved Network Handling and Added New Method `sendRequest`

## Description
This PR introduces significant improvements to the Vexana package, focusing on enhancing network handling and ensuring backward compatibility. The key changes include:

### 1. New Method `sendRequest`
**Reason:** The current `send` method, while functional, lacks the flexibility and robustness needed for more complex HTTP requests and responses.

- Introduced a new method `sendRequest` to provide a more robust and flexible way to send HTTP requests and handle responses. This method supports various HTTP methods, progress handling, cancellation, and caching.
- The `sendRequest` method aims to replace the older `send` method, offering enhanced functionality and cleaner code.

### 2. Deprecation of `send` Method
**Reason:** To ensure backward compatibility while encouraging users to migrate to the new and improved method.

- The `send` method has been marked as deprecated to encourage users to migrate to the new `sendRequest` method.
- The `send` method is still available for backward compatibility but will be removed in future versions.

### 3. NetworkResult Improvements
**Reason:** Enhance error handling and reduce unnecessary null checks, making the codebase cleaner and more maintainable.

- Changed `INetworkModel` from an abstract to a sealed class to facilitate pattern matching and reduce unnecessary null checks.
- Introduced `NetworkSuccessResult` and `NetworkErrorResult` to encapsulate success and error responses, respectively.
- Added utility methods for better handling of network responses.

### 4. Additional Methods
**Reason:** Provide utility methods to streamline response handling and error management, improving overall code quality and maintainability.

- Added helper methods like `fetchSuccessResponse`, `fetchErrorResponse`, `handleErrorResponse`, and `loadFromCache` to streamline response handling and error management.

### 5. Documentation
**Reason:** Ensure that users have clear guidance on how to use the new and modified methods.

- Updated documentation for all new and modified methods, providing clear guidance on their usage and parameters.

### 6. Unit Tests
**Reason:** Ensure code reliability and functionality by thoroughly testing new and modified methods.

- Added comprehensive unit tests for all new methods and significant changes, ensuring code reliability and functionality.

## Changes
- Added `sendRequest` method.
- Deprecated `send` method with clear documentation.
- Refactored `INetworkModel` to a sealed class.
- Implemented `NetworkSuccessResult` and `NetworkErrorResult`.
- Added utility methods for response and error handling.
- Updated and added new unit tests.

## Testing
- Comprehensive unit tests have been written and passed for all new and modified methods.
- Verified backward compatibility with existing code using the deprecated `send` method.

## Migration Guide
Users are encouraged to start using the `sendRequest` method. The `send` method is deprecated but still available for use. It will be removed in future versions, so migrating to `sendRequest` is recommended.

## Closing
These changes aim to enhance the functionality and maintainability of the Vexana package. Feedback and suggestions are welcome to further improve the package.

Thank you for considering this PR!

### CODE REVIEW

**Comment by MehmetKaranlik** [comment]
For deprecation case @VB10  need to decide, mb we want to keep both versions.

File: lib/src/model/network_result.dart [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  ///       print('Success: $data');
+  ///     },
+  ///     onError: (error) {
+  ///       print('Error: $error');
+  ///     },
+  ///   );
+  /// }
+  /// ```
+  B fold<B>({
+    required B Function(T data) onSuccess,
```
[the remark this answers is not in the document]
Comment (mysCod3r) [reply lib/src/model/network_result.dart]: Thank you for your response. I applied your advice, and I'm sure it will be more flexible.

**Review by MehmetKaranlik — APPROVED** [review]

**Comment by mysCod3r** [comment]
> @mysCod3r could you fix this conflict

Sure. resolved @VB10

**Comment by VB10** [comment]
@mysCod3r unit test has some issue could you check it again?

**Comment by mysCod3r** [comment]
> @mysCod3r unit test has some issue could you check it again?

I checked and saw that 'MockNetworkManager' was duplicated and Implementation missing in CustomNetworkManager. I fixed both.

**Review by VB10 — APPROVED** [review]
lgtmmmm

**Pull request merged**